include/nbl/asset/ICPUComputePipeline.h

```diff
@@ -61,6 +61,12 @@ class ICPUComputePipeline : public IComputePipeline<ICPUSpecializedShader, ICPUP
     }
     const ICPUPipelineLayout* getLayout() const { return m_layout.get(); }
 
+    inline void setLayout(core::smart_refctd_ptr<ICPUPipelineLayout>&& _layout)
+    {
+        assert(!isImmutable_debug());
+        m_layout = std::move(_layout);
+    }
+
```

Comment:
@Crisspl is this okay?

cause being able to change the layout of an already made descriptor set is a huge no-no (because of arrays that hold descriptors would need resizing etc.), wondering if there was a similar good reason why a pipeline didnt have a `setLayout` as well....